frontend/www/js/omegaup/components/contest/AddProblem.vue

```diff
@@ -195,15 +205,45 @@ export default class AddProblem extends Vue {
   publishedRevision: null | types.ProblemVersion = null;
   selectedRevision: null | types.ProblemVersion = null;
 
-  onSubmit(): void {
-    if (this.useLatestVersion) {
-      this.$emit('get-versions', {
-        target: this,
-        request: { problemAlias: this.alias },
-      });
-    } else {
-      this.onAddProblem();
+  setInitialRevisionValues(): void {
+    this.versionLog = [];
+    this.selectedRevision = this.publishedRevision = null;
+    this.useLatestVersion = true;
+  }
+
+  get problemsMapping(): MappedProblems {
+    let problemsMapping: MappedProblems = {};
+    for (const problem of this.problems) {
+      const publishedCommitHash = problem.commit;
+      problemsMapping[problem.alias] = {
+        versions: problem.versions.log,
+        mappedVersions: {},
+        published: problem.commit,
+        useLastVersion: false,
+      };
+      for (const [index, version] of problem.versions.log.entries()) {
+        problemsMapping[problem.alias].mappedVersions[version.commit] = version;
+        if (index === 0 && version.commit === publishedCommitHash) {
+          problemsMapping[problem.alias].useLastVersion = true;
+        }
+      }
```

Comment:
```suggestion
      const commitVersions = {};
      for (const [index, version] of problem.versions.log.entries()) {
        commitVersions[version.commit] = version;
      }
      problemMapping[problem.alias] = {
        problem,
        commitVersions,
      };
```

así no hay que pagar el precio de estar accediendo `problemsMapping[problem.alias]` tantas veces.

Reply:
Listo